### ISSUE TITLE
SIWE doesn't work with ERC-4337 (and pre-deployed contracts in general)

### ISSUE THREAD

**Issue by derekchiang** [spruceid/siwe#148]
Right now, SIWE uses ERC-1271 to validate signatures from contract accounts, but it only works if the contract is  already deployed.  If the contract is pre-deployed (aka "counterfactually deployed"), SIWE won't work.

Pre-deployed contract accounts are becoming increasingly common due to the popularity of ERC-4337, which doesn't deploy the contract account until the first transaction.  That was the motivation behind [EIP-6492](https://eips.ethereum.org/EIPS/eip-6492).  cc @Ivshti

TLDR: SIWE should implement EIP-6492.  Even though the EIP is not yet accepted, implementing 6492 won't break compatibility with existing wallets, and it will ensure that SIWE works with ERC-4337 which is becoming increasingly popular.

**Comment by Ivshti** [comment]
here's how to implement ERC-6492: https://github.com/AmbireTech/signature-validator/pull/3

**Comment by obstropolos** [comment]
Taking a look at this and also very excited for AA and Sign-In with Ethereum. We'll take a look at what compatibility would look like here - and thank you for the example @Ivshti !

**Comment by jd1900** [comment]
Not news on this?

**Comment by Ivshti** [comment]
here's something that may help integrating ERC-6492: https://github.com/spruceid/siwe/pull/153#issuecomment-1890390764

**Comment by wminshew** [comment]
is there any kind of timeline/roadmap available on when SIWE will support eips 4337/6492?

**Comment by Ivshti** [comment]
hey @obstropolos, any update?

**Comment by 0xTranqui** [comment]
just started working on a 6492 sign-in implementation and also find myself in this thread. really want to use SIWE but current lack of support means I need to fall back to use ambire tech or viems verifying libraries, which are great, but techincally dont match the SIWE spec. would love to see this

**Comment by jxom** [comment]
FWIW, 6492 is built into [Viem's SIWE implementation](https://viem.sh/docs/siwe/actions/verifySiweMessage). Would highly recommend using it. :D
